specs/NetworkSpec/RoutingTableExchangeAlgorithm.md

```diff
@@ -0,0 +1,213 @@
+- Proposal Name: New routing table exchange algorithm
+- Start Date: 6/21/2021
+- NEP PR: [nearprotocol/neps#0000](https://github.com/near/nearcore/pull/4112)
+- Issue(s): https://github.com/near/nearcore/issues/3838.
+- GitHub PR: https://github.com/near/NEPs/pull/220
+
+# Summary
+[summary]: #summary
+
+Currently, every node on the network stores its own copy of the [Routing Graph](NetworkSpec.md#Routing Table).
+The process of exchanging and verifying edges requires sending full copy of routing table on every synchronization.
+Sending full copy of routing table is wasteful, in this spec we propose a solution, which allows doing partial  synchronization.
+This should reduce both bandwidth used, and amount of CPU time.
+
+Typically, only one full sync would be required when a node connects to the network for the first time.
+For given two nodes doing synchronization, we propose a method, which requires bandwidth/processing time proportional to the size of data which needs to be exchanged for given nodes.
+
+# Algorithm
+In this spec we describe the implementation of `Routing Graph Reconciliation` algorithm using Inverse Bloom Filters.
+An overview of different `Routing Graph Reconciliation` algorithms can be a found at
+[ROUTING_GRAPH_RECONCILIATION.pdf](https://github.com/pmnoxx/docs/blob/piotr-test-markdown/near/ROUTING_GRAPH_RECONCILIATION.pdf)
+
+The process of exchanging `routing tables` involves exchanging edges between a pair of nodes.
+For given nodes `A`, `B`, that involves adding edges which `A` has, but `B` doesn't and vice-versa.
+For each connection we create a data structure [IbfSet](#IbfSet).
+It stores `Inverse Bloom Filters` [Ibf](#Ibf) of powers of `2^k` for `k` in range `10..17`.
+This allows us to recover around `2^17 / 1.3` edges with `99%` probability according to [Efficient Set Reconciliation without Prior Context](https://www.ics.uci.edu/~eppstein/pubs/EppGooUye-SIGCOMM-11.pdf).
+Otherwise, we do full routing table exchange.
+
+We have chosen a minimum size of `Ibf` to be `2^10`, because the overhead of processing `IBF` of smaller size of negligible, and to reduce the number of messages required in exchange.
+On the other side, we limit `Ibf` to size of `2^17` to reduce memory usage required for each data structure.
+
+
+# Routing Table
+d
```

Comment:
what happened here

Reply:
This should fix it: https://github.com/near/NEPs/pull/260/files